virtualenv.py

```diff
@@ -459,19 +465,35 @@ def _find_file(filename, folders):
     return False, filename
 
 
-def file_search_dirs():
-    here = os.path.dirname(os.path.abspath(__file__))
-    dirs = [here, join(here, "virtualenv_support")]
-    if os.path.splitext(os.path.dirname(__file__))[0] != "virtualenv":
-        # Probably some boot script; just in case virtualenv is installed...
+@contextlib.contextmanager
+def virtualenv_support_dirs():
+    """Context manager yielding either [virtualenv_support_dir] or []"""
+
+    # normal filesystem installation
+    if os.path.isdir(join(HERE, "virtualenv_support")):
+        yield [join(HERE, "virtualenv_support")]
```

Comment:
note this will still provide it if virtualenv support is a file instead of a folder (change of behavior compared to beforehand), we probably should warn in such cases and break instead

Reply:
it would always crash before if that were the case so it's not worth mucking this further:

```
$ rm -rf virtualenv_support/ && touch virtualenv_support && python virtualenv.py venv
New python executable in /tmp/virtualenv/venv/bin/python
Cannot find a wheel for setuptools
Cannot find a wheel for pip
Installing setuptools, pip, wheel...
  Complete output from command /tmp/virtualenv/venv/bin/python - setuptools pip wheel:
  Traceback (most recent call last):
  File "<stdin>", line 11, in <module>
ImportError: No module named pip
----------------------------------------
...Installing setuptools, pip, wheel...done.
Traceback (most recent call last):
  File "virtualenv.py", line 2344, in <module>
    main()
  File "virtualenv.py", line 712, in main
    symlink=options.symlink)
  File "virtualenv.py", line 948, in create_environment
    download=download,
  File "virtualenv.py", line 905, in install_wheel
    call_subprocess(cmd, show_stdout=False, extra_env=env, stdin=SCRIPT)
  File "virtualenv.py", line 796, in call_subprocess
    % (cmd_desc, proc.returncode))
OSError: Command /tmp/virtualenv/venv/bin/python - setuptools pip wheel failed with error code 1
```